src/components/forms/forms.js

```diff
@@ -0,0 +1,393 @@
+  const required = validators ? validators.findIndex(v => v === FormValidators.REQUIRED) !== -1 : false;
+
+  return div({
+    key: `formControl_${id}`,
+    style,
+    className: `formField-container formField-${id}`
+  }, [
+    title && !hideTitle && label({
+      id: `lbl_${id}`,
+      className: `control-label ${error ? 'errored' : ''}`,
+      htmlFor: `${id}`
+    }, [
+      title,
+      required && '*'
+    ]),
+    description && div({ style: { marginBottom: 15 } }, [description]),
+    formInput({
+      ...config,
+      error, setError,
+      formValue, setFormValue,
+      required
+    })
+  ]);
+};
+
+/*
+* Config options:
+* id
+* formFields: array[FormField Configs]
+* onChange, error, setError, formInfo, setFormInfo
+*/
+export const FormTable = (config) => {
+  const {
+    id, formFields, defaultValue,
+    onChange
+  } = config;
+
+  const [formValue, setFormValue] = useState(defaultValue || [{}]);
+
+  return div({ id, className: `formField-table formField-${id}` }, [
+    // generate columns
+    div({ className: 'formTable-row formTable-cols' }, formFields.map(x => {
+      return label({ className: 'control-label', id: `${id}-${x.title}` }, [x.title]);
+    })),
+    // generate form rows
+    formValue?.map((formRow, i) => {
+      return div({ className: 'formTable-row formTable-data-row', key: `formtable-${id}-${i}` }, [
+        ...formFields.map(formCol => {
+          return h(FormField, {
+            ...formCol,
+            id: `${id}-${i}-${formCol.id}`,
+            hideTitle: true, ariaDescribedby: `${id}-${formCol.title}`,
+            onChange: (({ value }) => onChange({key: `${id}.${i}.${formCol.id}`, value }))
+          });
+        }),
+        h(button, {
+          id: `delete-table-row-${id}-${i}`,
+          key: `delete-table-row-${id}-${i}`,
+          className: 'btn-formTable-delete btn-xs',
+          type: 'button',
+          onClick: () => {
+            const formValueClone = cloneDeep(formValue);
+            formValueClone.splice(i, 1);
+            setFormValue(formValueClone);
+            onChange({ key: id, value: formValueClone });
+          }
+        }, [
+          span({ className: 'glyphicon glyphicon-remove' })
+        ])
+      ]);
+    }),
+    // add new row to table button
+    div({ style: { display: 'flex', width: '100%', justifyContent: 'flex-end' } }, [
+      h(button, {
+        id: `add-new-table-row-${id}`,
+        key: `add-new-table-row-${id}`,
+        className: 'pill btn-xs',
+        style: {
+          backgroundColor: Theme.palette.secondary,
+          borderRadius: 0,
+          marginRight: 0
+        },
+        type: 'button',
+        onClick: () => {
+          const formValueClone = cloneDeep(formValue);
+          formValueClone.push({});
+          setFormValue(formValueClone);
+          onChange({ key: id, value: formValueClone });
+        }
+      }, [
+        'Add New Filetype',
```

Comment:
this string should be a prop so that the field form can be used generically

Reply:
done!